text/stacks/0004-jammy-jellyfish.md

```diff
@@ -0,0 +1,113 @@
+# Stacks based on Ubuntu 2022.04: Jammy Jellyfish
+
+## Summary
+
+A set of stacks based on the Ubuntu 2022.04 LTS (Jammy Jellyfish) release base
+image should be developed, released, and maintained by the Stacks subteam. Like
+the existing Bionic stacks, these new stacks should come in "full", "base", and
+"tiny" variants with similar, if not identical, sets of packages pre-installed.
+
+## Motivation
+
+Ubuntu provides a long-term-support (LTS) release every 2 years in April. These
+releases are supported by Canonical for 5 years. The current Bionic base image,
+first released in April 2018, will therefore be going out of support at the end
+of March 2023. In order to ensure that Paketo continues to provide a supported
+Ubuntu-based stack for our users, we should produce a new set of stack images,
+based on Jammy, alongside the current Bionic stacks.
+
+## Detailed Explanation
+
+### Variants
+
+The Jammy stack will be delivered in 3 variants that align to those already
+offered in the Bionic stack. The variants will be called `base`, `full`, and
+`tiny` just as they are in the Bionic stack. Each should be developed to offer
+roughly the same set of OS-level package support as their Bionic equivalent.
+
+### Stack IDs
+
+Each of these stacks will declare their stack ID as `io.paketo.stack.jammy`.
+This choice differs from Bionic in that we will **NOT** differentiate between
+the `tiny` and `base/full` variants using the stack ID.
+
+### User IDs
+
+These stacks will differ from Bionic with regards to their UID definitions.
+Each variant will ensure that the UID for the `build` phase is different than
+the `run` phase. This change will align more closely with the
+[recommendations](https://github.com/buildpacks/rfcs/blob/main/text/0085-run-uid.md)
+outlined in the Buildpacks Specification.
+
+### Image Naming and Tagging
+
+The stacks will name and tag their release images with the following pattern:
+
+```
+paketobuildpacks/jammy-{phase}-{variant}:{version}
+```
+
+For example we could see the following images for Jammy stacks:
+
+* `paketobuildpacks/jammy-build-base:latest`
```

Comment:
In my mind one drawback to pushing the differentiation into the repo name vs into the tag name is discoverability. For example, it's easy to go to https://hub.docker.com/_/ubuntu and, by reading the readme,  see what tags are available and how they relate. Yes, you could still go to the `paketobuildpacks` [org page](https://hub.docker.com/u/paketobuildpacks) and looking for stack repos, but there is a lot more noise to wade through and no opportunity to use a readme as a guidepost 

What if we did something like:
`paketobuildpacks/build:jammy-tiny-1.2.3`


Then, in the future we could even do things like repoint `:latest` from one ubuntu LTS to another (provided the CNB tooling eventually figures out how to preform a stack upgrade without generating bad images).

Reply:
What is `:latest` in this case? `jammy-base-latest`? What does this mean if we also have UBI stacks? We will soon be in a state where we have so many stacks and variants that something like a `:latest` tag becomes quite tricky to explain, let alone rely upon in a real usecase.

Reply:
As for discovery, I agree that we could be doing better. Barring a discovery of some sort of native solution in Dockerhub, could we consider modifications to the Paketo website to make this type of discovery easier?

Reply:
I've updated the RFC to outline that we should include a README on each stack repository that links out to the other variants to improve discoverability.

As for the underlying naming question, I still think the current RFC specification makes the most sense as a naming structure. It splits the stacks along their logical differences.

For some of the examples that had been proposed, like [Ubuntu](https://hub.docker.com/_/ubuntu), its possible to organize the images in the repository into a single linear ordering of versions. Each tag represents a different version of the same logical artifact, the Ubuntu OS distribution.

For other examples, like [Go](https://hub.docker.com/_/golang), its not clear that the choice to put everything in a single repository is a deliberate one or that it falls out of the constraints of there being no such thing as an organization within the top-level Dockerhub "library".